content/blog/announcing-python-tooling-improvements/index.md

```diff
@@ -0,0 +1,135 @@
+    acl="public-read",
+    website={
+        "indexDocument": "index.html",
+        "errorDocument": "error.html",
+    },
+    cors_rules=[{
+        "allowedHeaders": ["*"],
+        "allowedMethods": ["PUT", "POST"],
+        "allowedOrigins": ["https://s3-website-test.mydomain.com"],
+        "exposeHeaders": ["ETag"],
+        "maxAgeSeconds": 3000,
+    }],
+)
+```
+
+You would likely have several questions if writing the above example from scratch. What keys can be set for `website` and `cors_rules`? Should the keys be camelCase or snake_case? Which keys are required? What are the types of the values? To answer these questions, you'd need to refer to [`Bucket`]({{< relref "/docs/reference/pkg/aws/s3/bucket" >}})'s documentation.
+
+We now allow optionally passing nested values using data classes:
+
+```python
+from pulumi_aws import s3
+
+my_bucket = s3.Bucket("my-bucket",
+    acl="public-read",
+    website=s3.BucketWebsiteArgs(
+        index_document="index.html",
+        error_document="error.html",
+    ),
+    cors_rules=[s3.BucketCorsRuleArgs(
+        allowed_headers=["*"],
+        allowed_methods=["PUT", "POST"],
+        allowed_origins=["https://s3-website-test.mydomain.com"],
+        expose_headers=["ETag"],
+        max_age_seconds=3000,
+    )],
+)
+```
+
+IDEs can display argument names and types, reducing the need to refer to documentation. If a required argument is missing, or a value isn't the right type, IDEs will immediately show the problem, allowing errors to be caught earlier.
+
+The use of data classes makes setting nested values consistent with top-level values, with arguments that are always snake_case, for a more Pythonic feel.
+
+Here's what this looks like in VS Code using the new [Pylance](https://devblogs.microsoft.com/python/announcing-pylance-fast-feature-rich-language-support-for-python-in-visual-studio-code/) language server for Python:
+
+![VS Code](vscode-bucket.png)
+
+Support for passing data classes is a fully compatible change. Nested data can be passed as either raw `dict`s or as data classes, so existing programs using `dict`s will continue to work as-is without modification.
+
+## Accessing Outputs
+
+Nested outputs can now be accessed in a more strongly-typed way as well.
+
+Consider accessing one of the website settings from the bucket. Previously, nested outputs were returned as raw `dict`s and you'd access the value by looking it up in the `dict` by its key:
+
+```python
+index_document = my_bucket.website.apply(lambda w: w["indexDocument"] if w else None)
+```
+
+Now, nested outputs are returned as data classes and values can be accessed via properties.
+
+```python
+index_document = my_bucket.website.apply(lambda w: w.index_document if w else None)
+```
+
+Existing programs that lookup the values in the `dict` continue to work as-is, because the new output data classes are subclasses of `dict` and continue to return the same data with the same keys as before.
```

Comment:
```suggestion
Existing programs that lookup the values in the `dict` continue to work as-is because the new output data classes are subclasses of `dict` and continue to return the same data with the same keys as before.
```